library/nxos_command.py

```diff
@@ -0,0 +1,305 @@
+            command_list=dict(required=False),
+            text=dict(choices=BOOLEANS, type='bool'),
+            type=dict(choices=['show', 'config'], required=True),
+            protocol=dict(choices=['http', 'https'], default='http'),
+            host=dict(required=True),
+            username=dict(type='str'),
+            password=dict(type='str')
+        ),
+        required_one_of=[['command', 'command_list']],
+        mutually_exclusive=[['command', 'command_list']],
+        supports_check_mode=False
+    )
+    if not HAS_PYCSCO:
+        module.fail_json(msg='There was a problem loading pycsco')
+
+    auth = Auth(vendor='cisco', model='nexus')
+    username = module.params['username'] or auth.username
+    password = module.params['password'] or auth.password
+    protocol = module.params['protocol']
+    host = socket.gethostbyname(module.params['host'])
+
+    command = module.params['command']
+    command_list = module.params['command_list']
+    text = module.params['text'] or None
+    cmd_type = module.params['type'].lower()
+
+    device = Device(ip=host, username=username, password=password,
+                    protocol=protocol)
+
+    if isinstance(command, list):
+        module.fail_json(msg='Only single command are supported using "command". \
+            If you want to use a List, go for "command_list" instead.')
+
+    if cmd_type == 'show':
+        if command_list:
+            module.fail_json(msg='Only single show commands are supported. \
+                Lists are only supported for config changes \
+                at this time.')
+
+    if cmd_type == 'config':
+        if isinstance(command_list, str):
+            module.fail_json(msg='Only Lists are supported when "command_list" \
+                is used.')
+        if text:
+            module.fail_json(msg='Do not use text=true when using \
+                cmd_type=config. The text param is only \
+                supported when cmd_type=show')
+
+    if command:
+        commands = command
+    elif command_list:
+        commands = command_list
+
+    args = dict(commands=commands, text=text, cmd_type=cmd_type)
+    proposed = {}
+    changed = False
+    for param, value in args.iteritems():
+        if value is not None:
+            proposed[param] = value
+
+    if cmd_type == 'config':
+        if command:
+            if isinstance(command, str):
+                splitted_command = command.split(',')
+                cmds = command_list_to_string(splitted_command)
+        elif command_list:
+            cmds = command_list_to_string(command_list)
+
+    if cmd_type == 'show':
+        result = parsed_data_from_device(device, command, module, text)
+
+    elif cmd_type == 'config':
+        if cmds:
+            changed = True
+            result = send_command(device, cmds)
+
+    results = {}
+    results['changed'] = changed
+    results['proposed'] = proposed
+    results['commands'] = [commands]
```

Comment:
isn't this creating a list of lists for `commands`? We don't need that...too messy.

Instead, just `cmds` and return the command string from `command_list_to_string` - you'll still need to convert `command` to a list before calling that function though :)